include/rive/generated/animation/keyframe_bool_base.hpp

```diff
@@ -0,0 +1,70 @@
+#ifndef _RIVE_KEY_FRAME_BOOL_BASE_HPP_
+#define _RIVE_KEY_FRAME_BOOL_BASE_HPP_
+#include "rive/animation/keyframe.hpp"
+#include "rive/core/field_types/core_bool_type.hpp"
+namespace rive
+{
+	class KeyFrameBoolBase : public KeyFrame
+	{
+	protected:
+		typedef KeyFrame Super;
+
+	public:
+		static const uint16_t typeKey = 84;
```

Comment:
Storing it in some massive hashset is an option but it'd be significantly (at scale) slower than isTypeOf is currently. A switch statement in C++ will be evaluated as a direct memory lookup. We could store this in a contiguous array (with some waste when properties are missing) but I think letting the compiler optimize this will do better. We can definitely add the logic you're looking for with json mapping however, we could build it into a utility that the core generator builds for us.